katdal/applycal.py

```diff
@@ -307,103 +428,115 @@ def calc_correction_per_corrprod(dump, channels, cache, inputs,
     KeyError
         If input and/or cal product has no associated correction
     """
-    g_per_input = np.ones((len(inputs), len(channels)), dtype='complex64')
-    for product in cal_products:
-        for n, inp in enumerate(inputs):
-            sensor_name = 'Calibration/{}_correction_{}'.format(inp, product)
-            g_product = cache.get(sensor_name)[dump]
+    n_channels = channels.stop - channels.start
+    g_per_input = np.ones((len(params.inputs), n_channels), dtype='complex64')
```

Comment:
Personally I prefer `np.complex64` because it's at least theoretically possible for static analysis tools to tell you if you mis-spell it (although pylint seems very confused by numpy). I see katdal has a mixture, but that might be because you've used `'complex64'` and I've used `np.complex64`. I'll change the instance in this function to `'complex64'` to match what was already there.